frame/support/procedural/src/storage_alias.rs

```diff
@@ -406,15 +406,15 @@ impl Parse for StorageType {
 }
 
 /// The input expected by this macro.
-struct Input {
-	attributes: Vec<Attribute>,
-	visibility: Visibility,
+pub struct Input {
+	pub attributes: Vec<Attribute>,
+	pub visibility: Visibility,
 	_type: Token![type],
-	storage_name: Ident,
-	storage_generics: Option<SimpleGenerics>,
-	where_clause: Option<WhereClause>,
+	pub storage_name: Ident,
+	pub storage_generics: Option<SimpleGenerics>,
+	pub where_clause: Option<WhereClause>,
 	_equal: Token![=],
-	storage_type: StorageType,
+	pub storage_type: StorageType,
```

Comment:
strange to me these need to be pub now, where they were not needed to be pub before.

What was the specific reason?

Reply:
There doesn't seem to be any way other way to re-use this struct since it isn't public. The alternative would be to essentially copy-paste it which seems bad form to me. My usage of it is panicing if we try to attach `#[benchmarking(cached)]` to anything but a valid storage item declaration.

Reply:
would like @bkchr and or @KiChjang to confirm the approach here.

Not a big deal, but once you go pub, its hard to go back

Reply:
note: proc macro crates can only export proc macros, and this type is inside the `procedural` crate in frame support, so there is really no worry this will be used outside of frame proc macro land imo

Reply:
`pub(crate)` maybe

Reply:
AIUI the only reason why you need this type to be `pub` is because you need it to parse the `#[benchmark]` attribute. What if you parsed `#[pallet::whitelist]` instead?

Reply:
I'm fine with `pub (crate)` if this is satisfactory. Again since this is all happening within the `procedural` frame support crate, nothing but the proc macros can be exported or used elsewhere, so in theory we could mark every type in this crate as `pub` and it wouldn't really affect anything.

Reply:
I think the point here is that making these `pub` is a code smell, because we have not made them `pub` before and yet we were still able to parse pallet attributes.

Reply:
Yes, point taken, renaming the macro will also solve this issue and allow us to use our existing machinery for parsing `#[pallet::` macros